components/feature/webcompat-reporter/src/main/java/mozilla/components/feature/webcompat/reporter/WebCompatReporterFeature.kt

```diff
@@ -39,16 +49,15 @@ object WebCompatReporterFeature {
      * "android-components".
      */
     fun install(runtime: WebExtensionRuntime, productName: String = "android-components") {
-        runtime.installWebExtension(WEBCOMPAT_REPORTER_EXTENSION_ID, WEBCOMPAT_REPORTER_EXTENSION_URL,
+        extensionController.install(runtime,
             onSuccess = {
                 logger.debug("Installed WebCompat Reporter webextension: ${it.id}")
-                it.registerBackgroundMessageHandler(
-                    WEBCOMPAT_REPORTER_MESSAGING_ID,
+                extensionController.registerBackgroundMessageHandler(
```

Comment:
I think it's better to use `it.registerBackgroundMessageHandler` here, where `it` is the installed extension. Otherwise you're referencing  `this` (via `this.extensionController`) in the lambda. In other places this is problematic because we'd be leaking the feature instance. In your case this doesn't matter as your feature is a `object` (singleton), but I would still change it.

Reply:
So, it would just be `it.registerBackgroundMessageHandler(WEBCOMPAT_REPORTER_MESSAGING_ID, WebcompatReporterBackgroundMessageHandler(productName))`.

Reply:
Another alternative is to call `extensionsController.registerbackgroundMessageHandler()` before `extensionController.install` which will also work, as the the controller supports that, it will register once installed.